src/python/pants/backend/go/util_rules/go_bootstrap.py

```diff
@@ -25,101 +25,49 @@ class GoBootstrap:
 
 
 async def _go_search_paths(
-    env_tgt: EnvironmentTarget, golang_subsystem: GolangSubsystem, paths: Iterable[str]
+    env_tgt: EnvironmentTarget, golang_subsystem: GolangSubsystem, paths: Collection[str]
 ) -> tuple[str, ...]:
-
-    resolve_standard, resolve_local = "<ASDF>" in paths, "<ASDF_LOCAL>" in paths
-
-    if resolve_standard or resolve_local:
-        # AsdfToolPathsResult is uncacheable, so only request it if absolutely necessary.
-        asdf_result = await Get(
-            AsdfToolPathsResult,
-            AsdfToolPathsRequest(
-                env_tgt=env_tgt,
-                tool_name=golang_subsystem.asdf_tool_name,
-                tool_description="Go distribution",
-                resolve_standard=resolve_standard,
-                resolve_local=resolve_local,
-                paths_option_name="[golang].go_search_paths",
-                bin_relpath=golang_subsystem.asdf_bin_relpath,
-            ),
-        )
-        asdf_standard_tool_paths = asdf_result.standard_tool_paths
-        asdf_local_tool_paths = asdf_result.local_tool_paths
-    else:
-        asdf_standard_tool_paths = ()
-        asdf_local_tool_paths = ()
-
+    asdf_result = await AsdfToolPathsResult.get_un_cachable_search_paths(
+        paths,
+        env_tgt=env_tgt,
+        tool_name=golang_subsystem.asdf_tool_name,
+        tool_description="Go distribution",
+        paths_option_name="[golang].go_search_paths",
+        bin_relpath=golang_subsystem.asdf_bin_relpath,
+    )
     special_strings = {
-        "<ASDF>": lambda: asdf_standard_tool_paths,
-        "<ASDF_LOCAL>": lambda: asdf_local_tool_paths,
+        AsdfPathString.STANDARD.value: asdf_result.standard_tool_paths,
+        AsdfPathString.LOCAL.value: asdf_result.local_tool_paths,
     }
 
     expanded: list[str] = []
     for s in paths:
         if s == "<PATH>":
-            expanded.extend(await _environment_paths())
+            expanded.extend(await Get(PathEnvironmentVariable, {}))  # noqa: PNT30: Linear search
```

Comment:
Optional: You can clear out this `noqa` by fetching `PathEnvironmentVariable` outside of the loop. Obviously repeated `Get`s are only going to arise with pathological inputs, so 🤷

Reply:
Sure!